pkg/generate/app/pipeline.go

```diff
@@ -222,7 +222,7 @@ func AddServices(objects Objects) Objects {
 				}
 			}
 			if len(svc.Spec.Ports) == 0 {
-				glog.Warningf("DeploymentConfig %q: Cannot create a service with no ports", t.Name)
+				glog.Warningf("A service will not be generated for DeploymentConfig %q because no exposed ports were detected. Use 'osc expose dc %q --port=[port] --generator=service/v1' to create a service.", t.Name, t.Name)
```

Comment:
@kargakis @liggitt - I'll add a follow up pr to remove the quotes... I tried to think of an instance where we'd want them and I can't; the command is easier to read without them.